cascade-api/src/Cascade/Api/Data/Task.hs

```diff
@@ -0,0 +1,140 @@
+module Cascade.Api.Data.Task
+  ( Task
+  , Id
+  , Readable(..)
+  , RawCreatableV(..)
+  , RawCreatable
+  , ParsedCreatable(..)
+  , RawUpdatableV(..)
+  , RawUpdatable
+  , ParsedUpdatable(..)
+  , RawCreatableValidationErrors
+  , RawUpdatableValidationErrors
+  , parseRawCreatableTask
+  , parseRawUpdatableTask
+  )
+where
+
+import qualified Cascade.Api.Data.Id           as Data
+import           Data.Aeson                     ( FromJSON(..)
+                                                , ToJSON(..)
+                                                )
+import qualified Cascade.Api.Data.Project      as Project
+import           Cascade.Api.Data.OffsetDatetime
+                                                ( FormattedOffsetDatetime
+                                                , unFormattedOffsetDatetime
+                                                )
+import           Cascade.Api.Data.OffsetDatetime.Deadline
+                                                ( Deadline )
+import qualified Cascade.Api.Data.OffsetDatetime.Deadline
+                                               as Deadline
+import qualified Cascade.Data.Text             as Text
+import qualified Cascade.Data.Text.NonEmpty    as Text.NonEmpty
+import           Chronos                        ( Time )
+import           Cascade.Api.Data.Prelude
+import           Data.Generics.Labels           ( )
+import           Data.Monoid.Generic
+import           Validation
+
+data Task
+
+type Id = Data.Id Task
+
+data Readable = Readable
+  { id         :: Id
+  , title      :: Text.NonEmpty
+  , deadlineAt :: FormattedOffsetDatetime
+  , projectId  :: Project.Id
+  }
+  deriving stock (Generic, Show, Eq)
+  deriving anyclass (FromJSON, ToJSON)
+
+data RawCreatableV f = RawCreatable
+  { title      :: Validatable f Text (Maybe Text.NonEmpty.ValidationErrors)
+  , deadlineAt :: Validatable f FormattedOffsetDatetime (Maybe Deadline.ValidationErrors)
+  }
+  deriving stock Generic
+
+type RawCreatable = RawCreatableV Identity
+
+deriving stock instance Show RawCreatable
+deriving stock instance Eq RawCreatable
+deriving anyclass instance FromJSON RawCreatable
+deriving anyclass instance ToJSON RawCreatable
+
+type RawCreatableValidationErrors = RawCreatableV Validate
+
+deriving stock instance Show RawCreatableValidationErrors
+deriving anyclass instance FromJSON RawCreatableValidationErrors
+deriving anyclass instance ToJSON RawCreatableValidationErrors
+deriving via (GenericSemigroup RawCreatableValidationErrors) instance Semigroup RawCreatableValidationErrors
+deriving via (GenericMonoid RawCreatableValidationErrors) instance Monoid RawCreatableValidationErrors
```

Comment:
Do you think we can change this horrible thing that I had produced after merging #30?